lib/serverless.js

```diff
@@ -151,15 +155,21 @@ internals.writeFile = Util.promisify(Fs.writeFile);
 
 internals.rimraf = Util.promisify(Rimraf);
 
-internals.getServer = async (root) => {
+internals.getServer = async (rootOrPath) => {
+
+    let path = Path.join(rootOrPath, 'server');
 
-    const path = Path.join(root, 'server');
+    try {
+        await stat(rootOrPath.endsWith('js') ? rootOrPath : `${rootOrPath}.js`);
+        path = rootOrPath;
+    }
+    catch (err) { }
```

Comment:
Unfortunately we can never completely ignore errors— the closest we get is this using [bounce](https://github.com/hapijs/bounce):
```js
try {
    await something();
}
catch (err) {
    Bounce.rethrow(err, 'system');
}
```

In this case it is a moot point because I believe we don't need to handle the special case: `require(path)` will run the [module resolution algorithm](https://nodejs.org/api/modules.html#modules_all_together) on `path`, which already normalizes the file extension away.  If for some reason that is not doing the trick, I recommend using `require.resolve()` to normalize the path.

Reply:
@devinivy, I really appreciate these thorough explanations of why and your expectations.  It helps ALOT.

I've replaced the above section with this.  

```
    let path = Path.join(rootOrPath, 'server');

    try {
        path = require.resolve(rootOrPath);
    }
    catch (err) { 
        Bounce.rethrow(err, 'system');
    }
```